### PR TITLE
Sc 276 connect frontend with endpoint create task

### PR DESCRIPTION
**Pull request by marialobillo**
Created create task service on the client side and added cors on server.

TODO: setup/fix dotenv file for reactjs+typescript
Screenshot 2024-06-01 at 07 48 02

### CODE REVIEW

File: client/src/env.d.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+declare namespace NodeJS {
+  interface ProcessEnv {
+    NODE_APP_API_URL: string;
+  }
+}
```
Comment (AbdelrahmanRadwan) [inline client/src/env.d.ts]: Did you want to ignore this file? why do you need it? Also its name is a bit strange.

File: client/src/pages/CreateTask.tsx [hunk 2]
```diff
@@ -43,20 +39,30 @@ const CreateTask = () => {
...
+          setError('An unexpected error occurred.');
+        }
+      }
     }
   }
 
 
   return (
     <>
+      {error && <p style={{ color: 'red' }}>{error}</p>}
```
Comment (AbdelrahmanRadwan) [inline client/src/pages/CreateTask.tsx]: I would recommend moving the styling to a css class in a separate file.

File: client/vite.config.ts [hunk 3]
```diff
@@ -1,7 +1,15 @@
...
 export default defineConfig({
   plugins: [react()],
+  define: {
+    "process.env": {},
+  },
 })
+
+
+
+
```
Comment (AbdelrahmanRadwan) [inline client/vite.config.ts:12]: Can remove this.

File: server/.gitignore [hunk 4]
```diff
@@ -128,3 +128,6 @@ dist
 .yarn/build-state.yml
 .yarn/install-state.gz
 .pnp.*
+
+
```
Comment (AbdelrahmanRadwan) [inline server/.gitignore]: Can remove.

File: client/src/env.d.ts (hunk 1 above)
Comment (marialobillo) [reply client/src/env.d.ts]: Yeah, I was trying to setup .env on reactjs+typescript but is not working

File: client/vite.config.ts (hunk 3 above)
Comment (marialobillo) [reply client/vite.config.ts:12]: Thanks, done

File: server/.gitignore (hunk 4 above)
Comment (marialobillo) [reply server/.gitignore]: Do you mean the 'build'?

Comment (AbdelrahmanRadwan) [reply server/.gitignore]: I meant the empty lines.

**Review by AbdelrahmanRadwan — APPROVED** [review]

**Pull request merged**